### DISCUSSION TITLE
Skip login page, when using middleware.

### DISCUSSION THREAD

**Discussion by atissedredecathlon** [nextauthjs/next-auth#4511] [Ideas]
### Description 📓

Hey, 

The logic to directly access a provider and skip the login page [is perfectly clear in the docs ](https://next-auth.js.org/getting-started/client#starts-oauth-sign-in-flow-when-clicked), when using the client-side hook. 

However, for those who implemented the authentication check with the middleware, there isn'n a clean solution. 

The only way to do that seems to be recreating an entire login page (which will never be seen by users) and implement the front-end solution from the docs here.

It seems to me that there should be a cleaner way to do that, especially since the middleware one-liner solution is so simple and elegant : `export { default } from 'next-auth/middleware';`

### How to reproduce ☕️

An example of what it could like in the providers config ? 
```
providers: [
    {
      default: true, <-- skips login page
      id: 'customProvider',
      name: 'Custom Provider',
      type: 'oauth',
      ...
    },
  ],
```

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by ghost** [comment]
I am also in need of such as feature as middleware is so elegant - shame it forces a redundant login page

**Comment by always-maap** [comment]
not the best option but at least it works

https://github.com/nextauthjs/next-auth/discussions/4078#discussioncomment-2270608

**Reply by always-maap** [reply]
but still, I really wanted this feature

**Comment by robertveloso** [comment]
+1

**Comment by scerelli** [comment]
+1 

especially now with server components this is something really important to have in next auth. I think also something quite simple, like: `redirect('/api/auth/signin/auth0')` might be enough

**Comment by federicocappellotto97** [comment]
+1

**Comment by bensonjm** [comment]
+1

**Comment by hobadams** [comment]
+1

**Comment by sebsch1337** [comment]
+1

**Comment by Nx5** [comment]
+1

**Comment by dimamarkus** [comment]
+1

**Comment by IgorKomissarov** [comment]
+1

**Comment by anirudhv** [comment]
+1

**Comment by PandaPandaDev** [comment]
+1

**Comment by Firgrep** [comment]
+1

**Comment by quickbeard** [comment]
+1

**Comment by isoroka-plana** [comment]
+1

**Comment by arielff3** [comment]
+1

**Comment by ricodealma** [comment]
+1

**Comment by Gaks3** [comment]
+1

**Comment by BehnamAxo** [comment]
+1

**Comment by thisizkp** [comment]
This is an important feature. Yes. But there's no point of spamming the thread with `+1` requests. Please avoid doing that.

**Comment by jphillips28** [comment]
+1

**Comment by AchyutPL** [comment]
```
import { withAuth } from 'next-auth/middleware';
import { NextResponse } from 'next/server';
import { IF_AUTHENTICATED_DEFAULT_REDIRECT, PUBLIC_URLS_THAT_IS_NOT_ALLOWED_IF_AUTHENTICATED } from './config';

const isDevMode = process.env.NODE_ENV !== 'production';

const auth_Middleware = withAuth(async (req) => {
    console.log('🚀🚀🚀🚀Middleware!!!🚀🚀🚀🚀');

    const isNotAllowedPublicRouteIfAuthenticated = PUBLIC_URLS_THAT_IS_NOT_ALLOWED_IF_AUTHENTICATED.includes(req.nextUrl.pathname);

    if (isNotAllowedPublicRouteIfAuthenticated) {
        return NextResponse.redirect(new URL(IF_AUTHENTICATED_DEFAULT_REDIRECT, req.url));
    }

    return NextResponse.next();
},
    {
        pages: {
            signIn: '/auth/sign-in',
        }
    }
);

const default_Middleware = (req: Request) => {
    return NextResponse.next();
};


const middleware = isDevMode ? default_Middleware : auth_Middleware;

export default middleware;

export const config = { matcher: ['/lancozen/:path*'] };

**Comment by agildw** [comment]
+1
